docs/codeql/codeql-cli/about-codeql-workspaces.rst

```diff
@@ -0,0 +1,94 @@
+* The ``ignore`` block contains a list of glob patterns that define CodeQL packs that are not available in the workspace.
+
+Each entry in the ``provide`` or ``ignore`` section must map to the location of a ``qlpack.yml`` file. All glob patterns are defined relative to the directory that contains the workspace file. For a list of patterns accepted in this file, see "`@actions/glob <https://github.com/actions/toolkit/tree/main/packages/glob#patterns>`__ ."
+
+For example, the following ``codeql-workspace.yml`` file defines a workspace that contains all the CodeQL packs recursively found in the ``codeql-packs`` directory, except for the packs in the ``experimental`` directory:
+
+.. code-block:: yaml
+
+    provide:
+      - "*/codeql-packs/**/qlpack.yml"
+    ignore:
+      - "*/codeql-packs/**/experimental/**/qlpack.yml"
+
+To verify that your ``codeql-workspace.yml`` file includes the CodeQL packs that you expect, run the ``codeql pack ls`` command in the same directory as your workspace. The result of the command is a list of all CodeQL packs in the workspace.
+
+
+CodeQL workspaces and query resolution
+--------------------------------------
+
+All CodeQL packs in a workspace are available as source dependencies for each other when you run any CodeQL command that resolves queries or packs. For example, when you run ``codeql pack install`` in a pack directory in a workspace, any dependency that can be found in the workspace will be used instead of downloading that dependency to the package cache and adding it to the ``codeql-pack.lock.yml`` file. For more information, see ":ref:`Adding and Installing Dependencies <adding-and-installing-dependencies>`."
+
+Similarly, when you publish a CodeQL query pack to the GitHub container registry using  ``codeql pack publish`` the command will always use the dependencies from the workspace instead of using dependencies found in the local package cache.
+
+This ensures that any local changes you make to a query library in a dependency are automatically reflected in any query packs you publish from that workspace.
+
+.. pull-quote::
+
+  Note
+
+  Source dependencies are CodeQL packs that are resolved from the local file system. These dependencies can be in the same CodeQL workspace, or specified as a path option using the ``--additional-packs`` argument. When you compile and run queries locally, source dependencies override any dependencies found in the local package cache as well as version constraints defined in the ``qlpack.yml``.
+
+  This is particularly useful in the following situations:
+
+  - One of the dependencies of the query pack you are running is not yet published. Resolving from source is the only way to reference that pack.
+  - You are making changes to multiple packs at the same time and want to test them together. Resolving from source ensures that you are using the version of the pack with your changes in it.
```

Comment:
I should have spotted this in my original review. It seems as if this should be a section with its own title somewhere in this article. On line 10 we introduce the term in italics, but we don't currently have anyway to link to more information about source dependencies.

I'd recommend moving this information into a new section called "Source dependencies" or "Source dependencies in a workspace". Exactly where this section should go, and what other changes we should make to the rest of the content isn't entirely clear to me. It seems as if this concept is fundamental to the purpose of a workspace, so could arguably go towards the top of the article.

Reply:
Thank you. That is a good suggestion. I had a stab at this. I decided to put the source dependencies immediately after the section describing the `codeql-workspace.yml` file itself. I think this works ok.

Reply:
👍🏻 Thanks 🙂